pulp_ansible/app/viewsets.py

```diff
@@ -436,6 +437,57 @@ def sign(self, request, pk):
         )
         return OperationPostponedResponse(result, request)
 
+    def _handle_copy_or_move(self, request, copy_or_move):
+        repository = self.get_object()
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data = serializer.validated_data
+
+        dest_repos_pks = [x.pk for x in data["destination_repositories"]]
+
+        signing_service = data.get("signing_service", None)
+        if signing_service:
+            signing_service = signing_service.pk
+
+        reserved = data["destination_repositories"] + [repository]
```

Comment:
Is the signing the reason to make the source repository exclusive?

Reply:
Yes. We could check the serializer and modify that if no signing service is found